course_content/notebooks/dask_intro.ipynb

```diff
@@ -1241,6 +1241,314 @@
+   "source": [
+    "### Iris in-built lazy operations : statistics\n",
+    "\n",
+    "Let's take average winds over the region.  \n",
+    "This uses the Iris Cube.collapsed() operation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mean_u = u_cube.collapsed(('latitude','longitude'), iris.analysis.MEAN)\n",
+    "mean_v = v_cube.collapsed(('latitude','longitude'), iris.analysis.MEAN)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As the input cubes have lazy data, and the MEAN operator is lazy-preserving, the results' data should still be deferred calculations -- i.e. the source data has not yet been loaded from disk.\n",
+    "\n",
+    "Check this out, as an exercise ..."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Exercise: check lazy status\n",
+    "\n",
+    "**1.** Show that the above results have not yet been calculated."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**2.** Print the mean wind values "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**3.**  \n",
+    "Have the data been loaded ?  \n",
+    "Are the mean cubes still lazy ?  \n",
+    "What about the original u/v cubes ?  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Iris in-built lazy operations : cube arithmetic\n",
+    "\n",
+    "We can easily calculate windspeeds as a vector magnitude.  \n",
+    "This can be made from the U and V cubes using the Iris \"cube arithmetic\" operators.  \n",
+    "The result is still lazy -- i.e. data has not yet been loaded from disk."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cube_windspeed = (u_cube * u_cube + v_cube * v_cube) ** 0.5\n",
+    "cube_windspeed.rename('wind_speed')\n",
+    "\n",
+    "print 'Windspeed cube:'\n",
+    "print cube_windspeed\n",
+    "print\n",
+    "print 'Windspeed cube : lazy = ', cube_windspeed.has_lazy_data()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Custom lazy calculations\n",
+    "\n",
+    "Suppose though that we want to calculate wind _**directions**_.  \n",
+    "\n",
+    "This is somewhat harder, as Iris doesn't provide a function for the calculation.  \n",
```

Comment:
This is somewhat harder ~~,~~ as Iris doesn't provide a function for **this** calculation.